examples/auction/src/lib.rs

```diff
@@ -3,206 +3,191 @@ use core::fmt::Debug;
-/// Bids are to be placed before the auction end. After that, bids are refused.
-/// Only bids that exceed the highest bid are accepted.
-/// Bids are placed incrementally, i.e., an account's bid is considered
-/// to be the **sum** of all bids.
+/// Users can call the bid function to participate in the auction.
+/// A user has to send some CCD when calling the bid function.
+/// This CCD amount has to exceed the current highest bid to be accepted by the
+/// smart contract.
 ///
-/// Example: if Alice first bid 1 CCD and then bid 2 CCD, her total
-/// bid is 3 CCD. The bidding will only go through if 3 CCD is higher than
-/// the currently highest bid.
-///
-/// After the auction end, any account can finalize the auction.
-/// The auction can be finalized only once.
-/// When the auction is finalized, every participant except the
-/// winner gets their money back.
+/// The smart contract keeps track of the current highest bidder as well as
+/// the CCD amount of the highest bid. The CCD balance of this smart contract
+/// represents the highest bid. When a new highest bid is accepted by the smart
+/// contract, the smart contract refunds the old highest bidder.
+
+/// Bids have to be placed before the auction ends. The participant with the
+/// highest bid (the last bidder) wins the auction.
+
+/// After the auction ends, any account can finalize the auction. The owner of
+/// this contract receives the highest bid (the balance of this contract) when
+/// the auction is finalized. This can be done only once.
 
-/// The state in which an auction can be.
+/// The state of the auction.
 #[derive(Debug, Serialize, SchemaType, Eq, PartialEq, PartialOrd, Clone)]
 pub enum AuctionState {
     /// The auction is either
     /// - still accepting bids or
     /// - not accepting bids because it's past the auction end, but nobody has
     ///   finalized the auction yet.
     NotSoldYet,
-    /// The auction is over and the item has been sold to the indicated address.
-    Sold(AccountAddress), // winning account's address
+    /// The auction has been finalized and the item has been sold to the
+    /// winning `AccountAddress`.
+    Sold(AccountAddress),
 }
 
 /// The state of the smart contract.
-/// This is the state that will be shown when the contract is queried using
+/// This state can be viewed by querying the node with the command
 /// `concordium-client contract show`.
-#[derive(Debug, Serial, DeserialWithState)]
-#[concordium(state_parameter = "S")]
-pub struct State<S> {
+#[derive(Debug, Serial, Deserial)]
+pub struct State {
     /// The part of the state that can be viewed
     viewable_state: ViewableState,
-    /// Keeping track of which account bid how much money
-    bids:           StateMap<AccountAddress, Amount, S>,
 }
 
-/// The part of the state to be viewed using `concordium-client contract invoke`
+/// This part of the state can be viewed by querying the node
+/// with the command `concordium-client contract show`.
```

Comment:
`concordium-client contract show` does not include the state/model for v1 contracts. The idea, instead, is to use `contract invoke` to invoke entrypoints that return values to be viewed.